### PR TITLE
try dask.array

### PR DESCRIPTION
**Pull request by aaronspring**
- tries to fix https://github.com/xgcm/xhistogram/issues/70, not successful
- speeds up CI

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "Topic :: Scientific/Engineering",
 ]
 
-INSTALL_REQUIRES = ["xarray>=0.12.0", "dask", "numpy>=1.17"]
+INSTALL_REQUIRES = ["xarray>=0.12.0", "dask.array", "numpy>=1.17"]
```
Comment (rabernat) [inline setup.py:23]:
```suggestion
INSTALL_REQUIRES = ["xarray>=0.12.0", "dask[array]", "numpy>=1.17"]
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @aaronspring! FWIW I'm not sure this will make an large impact on what dependencies are installed. Dask's required dependencies are listed here

https://github.com/dask/dask/blob/c19a8300d514a8ee488e0da5e81c01155d55fdbe/setup.py#L32-L39

which are always installed, regardless if one `pip install`s `dask`, `dask[array]`, `dask[array, dataframe]`, `dask[complete]`, etc. 

Adding the `[array]` extra to the `dask` install will cause `numpy` to also be installed

https://github.com/dask/dask/blob/c19a8300d514a8ee488e0da5e81c01155d55fdbe/setup.py#L13

This is probably a good thing as it will require `xhistogram` to install a `numpy` version which is compatible with `dask.array`, but will, as of today, not result in a big difference since `numpy` is listed as a separate dependency already. 

That's all to say that this change seems totally reasonable, but isn't the cause for libraries like `bokeh` being installed as referenced over in https://github.com/xgcm/xhistogram/issues/70

**Comment by aaronspring** [comment]
thanks @jrbourbeau for the quick description

ok. so there is no way around `install_requires` and the other extras (`bokeh` in diagnostics) were not even installed before.

so this PR here would only ensure that a matching `numpy` version is used @rabernat but `xarray` already kind of does that by itself. should this PR still go forward?

File: ci/environment-3.7.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ channels:
 dependencies:
   - python=3.7
   - xarray
-  - dask
+  - dask[array]
```
Comment (jrbourbeau) [inline ci/environment-3.7.yml:7]: Ah, I see what the issue might be now. `dask[array]` (and my comment here https://github.com/xgcm/xhistogram/pull/71#pullrequestreview-810615021) is just a `pip` thing and isn't supported when installing `dask` with `conda`. `conda install dask` will pull in other packages like `bokeh` though. Specifically the packages listed [here](https://github.com/conda-forge/dask-feedstock/blob/98bc1b513a8df48ef1cf5c10f1d20eea61ea9691/recipe/meta.yaml#L20-L27). 

If you just want to `conda install` the minimum required dependencies for Dask (which would be equivalent to `pip install dask`), you'll want to install the `dask-core` package from `conda-forge`.

Comment (aaronspring) [reply ci/environment-3.7.yml]: using `dask-core` reduces `Setup Conda Env` from 5:37 https://github.com/xgcm/xhistogram/runs/4258344230?check_suite_focus=true to 2:47 https://github.com/xgcm/xhistogram/runs/4258767295?check_suite_focus=true

**Comment by aaronspring** [comment]
thanks for the explanation. the originating issue was https://github.com/xarray-contrib/xskillscore/issues/359, then we discovered that `xhistogram` relies on `dask`.

also `xhistogram` on `conda-forge` relies on `dask` https://github.com/conda-forge/xhistogram-feedstock/blob/master/recipe/meta.yaml - not sure whether there also specifying `dask.array` works, but you said `dask[array]` is a `pip` thing, so likely not

**Comment by rabernat** [comment]
If we are happy about this I will merge. :+1: if you think we are done here.

**Comment by aaronspring** [comment]
happy to merge and maybe we can ship the `conda` version with `dask-core`, see https://github.com/conda-forge/xhistogram-feedstock/pull/9

**Pull request merged**